app/models/spree/product.rb

```diff
@@ -78,8 +78,7 @@ class Product < ApplicationRecord
         where(producer_properties: { property_id: property_ids }).
         or(
           where(spree_product_properties: { property_id: property_ids })
-        ).
-        distinct
```

Comment:
I'm a bit worried about the implications here. The `distinct` was added because the scope uses `left_outer_joins` and that can multiply the product rows if a product has multiple properties.

The sorting query doesn't seem to use the joins. It uses grouping instead. But the non-sorting query could now produce too many rows. Did you test that?

Reply:
I've _tested_ by changing the implementation of `spec/controllers/api/v0/order_cycles_controller_spec.rb`:

```ruby
expect(product_ids).to eq [product1.id, product2.id]
```
instead of
```ruby
expect(product_ids).to include product1.id, product2.id
```

I thought it was enough, but maybe I'm wrong.

Reply:
I see. I had another look and when I call `with_properties` with multiple properties and a product has several of them then the product appears several times. But ransack seems to add the grouping which de-duplicates the records and everything is fine.

At the moment, this scope is only used with ransack and only in this one controller, I think. And it works here. We just have to be aware of this if we ever try to use this scope in a different context.